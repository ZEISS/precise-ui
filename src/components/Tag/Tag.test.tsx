import * as React from 'react';
import * as enzyme from 'enzyme';
import { Tag } from './';

describe('<Tag />', () => {
  it('should render an empty <Tag> component', () => {
    const wrapper = enzyme.shallow(<Tag />);
    const nodes = wrapper.filter((node: any) => node.text() !== '');
    expect(nodes.exists()).toBeFalsy();
  });

  it('should render a <Tag> with a primary type if nothing given', () => {
    const wrapper = enzyme.mount(<Tag>my-nice-tag</Tag>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Tag> with a primary type if something wrong given', () => {
    // @ts-ignore: test case
    const wrapper = enzyme.mount(<Tag type="bar">foo</Tag>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Tag> with a info type if explicitly given', () => {
    const wrapper = enzyme.mount(<Tag type="info">foo</Tag>);
    expect(wrapper).toMatchSnapshot();
  });
});
