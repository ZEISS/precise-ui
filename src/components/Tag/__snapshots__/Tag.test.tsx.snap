// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render a <Tag> with a info type if explicitly given 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  font-size: 0.8em;
  border: 0;
  color: rgba(47,101,20,1);
  background-color: rgba(201,239,148,1);
  padding: 0.25rem 1rem;
}

<Tag
  type="info"
>
  <styled.span
    theme={
      Object {
        "tagBackground": "rgba(201,239,148,1)",
        "tagColor": "rgba(47,101,20,1)",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-block;
  border-radius: 3px;
  font-size: 0.8em;
  border: 0;
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  padding: ",
              "0.25rem",
              " ",
              "1rem",
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "tagBackground": "rgba(201,239,148,1)",
          "tagColor": "rgba(47,101,20,1)",
        }
      }
    >
      <span
        className="c0"
      >
        foo
      </span>
    </StyledComponent>
  </styled.span>
</Tag>
`;

exports[`<Tag /> should render a <Tag> with a primary type if nothing given 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  font-size: 0.8em;
  border: 0;
  color: rgba(52,93,127,1);
  background-color: rgba(193,230,254,1);
  padding: 0.25rem 1rem;
}

<Tag>
  <styled.span
    theme={
      Object {
        "tagBackground": "rgba(193,230,254,1)",
        "tagColor": "rgba(52,93,127,1)",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-block;
  border-radius: 3px;
  font-size: 0.8em;
  border: 0;
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  padding: ",
              "0.25rem",
              " ",
              "1rem",
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "tagBackground": "rgba(193,230,254,1)",
          "tagColor": "rgba(52,93,127,1)",
        }
      }
    >
      <span
        className="c0"
      >
        my-nice-tag
      </span>
    </StyledComponent>
  </styled.span>
</Tag>
`;

exports[`<Tag /> should render a <Tag> with a primary type if something wrong given 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  font-size: 0.8em;
  border: 0;
  color: rgba(52,93,127,1);
  background-color: rgba(193,230,254,1);
  padding: 0.25rem 1rem;
}

<Tag
  type="bar"
>
  <styled.span
    theme={
      Object {
        "tagBackground": "rgba(193,230,254,1)",
        "tagColor": "rgba(52,93,127,1)",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-block;
  border-radius: 3px;
  font-size: 0.8em;
  border: 0;
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  padding: ",
              "0.25rem",
              " ",
              "1rem",
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "tagBackground": "rgba(193,230,254,1)",
          "tagColor": "rgba(52,93,127,1)",
        }
      }
    >
      <span
        className="c0"
      >
        foo
      </span>
    </StyledComponent>
  </styled.span>
</Tag>
`;
